azalea-chat/src/component.rs

```diff
@@ -89,41 +135,26 @@ impl FormattedText {
     /// println!("{}", component.to_ansi());
     /// ```
     pub fn to_ansi(&self) -> String {
-        // default the default_style to white if it's not set
-        self.to_ansi_with_custom_style(&DEFAULT_STYLE)
+        self.to_custom_format(
+            |running, new, default| (running.compare_ansi(new, default), "".to_owned()),
+            |text| text.to_string(),
+            |style| if !style.is_empty() { "\u{1b}[m" } else { "" }.to_string(),
+            &DEFAULT_STYLE,
+        )
     }
 
-    /// Convert this component into an
-    /// [ANSI string](https://en.wikipedia.org/wiki/ANSI_escape_code).
-    ///
-    /// This is the same as [`FormattedText::to_ansi`], but you can specify a
-    /// default [`Style`] to use.
-    pub fn to_ansi_with_custom_style(&self, default_style: &Style) -> String {
-        // this contains the final string will all the ansi escape codes
-        let mut built_string = String::new();
-        // this style will update as we visit components
-        let mut running_style = Style::default();
-
-        for component in self.clone().into_iter() {
-            let component_text = match &component {
-                Self::Text(c) => c.text.to_string(),
-                Self::Translatable(c) => c.to_string(),
-            };
-
-            let component_style = &component.get_base().style;
-
-            let ansi_text = running_style.compare_ansi(component_style, default_style);
-            built_string.push_str(&ansi_text);
-            built_string.push_str(&component_text);
-
-            running_style.apply(component_style);
-        }
-
-        if !running_style.is_empty() {
-            built_string.push_str("\u{1b}[m");
-        }
-
-        built_string
+    pub fn to_html(&self) -> String {
```

Comment:
There should probably be at least one test for this (see the to_ansi tests in `text_component.rs` and `integration_test.rs`)